clustered/client/src/main/java/org/ehcache/clustered/client/internal/PerpetualCachePersistenceException.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal;
+
+import org.ehcache.CachePersistenceException;
+
+/**
+ * Thrown to indicate a perpetual (non-transient) failure in a persistent cache manager.
+ * <p>
+ * Receiving this exception indicates that future interactions with the throwing entity will continue to fail without
+ * corrective action.
+ *
+ * @see CachePersistenceException
+ */
+public class PerpetualCachePersistenceException extends CachePersistenceException {
```

Comment:
I would prefer an exception name that better reflect the fact that the persistent layer cannot be used anymore instead of the fact that this exception will be throwing again and again. Something like: `UnusableCachePersistenceException` or `InvalidCachePersistenceException` or `IllegalStateCachePersistenceException`...

Reply:
At the point of throwing this exception that's not really true. The persistence layer could be used again if someone created the relevant entities that are missing. This exception signals to the outer layers of the clustering code that we're hitting a non-transient issue with the persistence code. It chooses to translate that it to a 'graveyard' state where nothing can be made to work again.  I think your alternative exception names might fit better for the exceptions thrown from the `FailedReconnectServerStoreProxy` that gets installed.

Reply:
cool!
So that means that we could have a config in Ehcache like "autoRecover(true|false)" that would allow Ehcache to automatically recreate the persistence layer then ?

Reply:
That would likely not be a distinct setting but more likely a facet of the behavior of the existing 'autoCreate' option. If the cache isn't auto create already it's either not the right thing  to do to create on reconnect, or we don't even have the necessary information to do the create.